core/src/main/java/com/sequenceiq/cloudbreak/repository/StackRepository.java

```diff
@@ -6,18 +6,21 @@
 import javax.transaction.Transactional;
 
 import org.springframework.data.jpa.repository.Query;
-import org.springframework.data.repository.CrudRepository;
 import org.springframework.data.repository.query.Param;
 
 import com.sequenceiq.cloudbreak.api.model.Status;
+import com.sequenceiq.cloudbreak.aspect.BaseRepository;
 import com.sequenceiq.cloudbreak.domain.Credential;
 import com.sequenceiq.cloudbreak.domain.FlexSubscription;
 import com.sequenceiq.cloudbreak.domain.Network;
 import com.sequenceiq.cloudbreak.domain.stack.Stack;
+import com.sequenceiq.cloudbreak.aspect.HasPermission;
+import com.sequenceiq.cloudbreak.service.EntityType;
 
 @EntityType(entityClass = Stack.class)
 @Transactional(Transactional.TxType.REQUIRED)
-public interface StackRepository extends CrudRepository<Stack, Long> {
+@HasPermission
```

Comment:
I think this will break HA as there are many stack related queries from the Heartbeatservice

Reply:
If there is no security context then it allows to go through so it will work. Ignore the comment.